extensions/azure/cosmos/contract-negotiation-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/negotiation/store/CosmosContractNegotiationStoreIntegrationTest.java

```diff
@@ -316,43 +297,37 @@ void nextForState_leasedBySelf() {
 
         // let's verify that the first invocation correctly sets the lease
         var result = store.nextForState(state.code(), 10);
-        assertThat(result).hasSize(1); //should contain the lease already
+        assertThat(result).hasSize(1).extracting(ContractNegotiation::getId).containsExactly(n.getId()); //should contain the lease already
         var storedNegotiation = readItem(n.getId());
         assertThat(storedNegotiation.getLease()).isNotNull().hasFieldOrPropertyWithValue("leasedBy", CONNECTOR_ID);
 
         // verify that the subsequent call to nextForState does not return the entity
         result = store.nextForState(state.code(), 10);
-
         assertThat(result).isEmpty();
     }
 
     @Test
-    void nextForState_leaseByAnotherExpired() throws InterruptedException {
+    void nextForState_leaseByAnotherExpired() {
         var state = ContractNegotiationStates.CONFIRMED;
         var n = generateNegotiation(state);
         var doc = new ContractNegotiationDocument(n, partitionKey);
-        doc.acquireLease("another-connector", Duration.ofMillis(10));
-        container.createItem(doc);
-
-        Thread.sleep(20); //give the lease time to expire
-
-        var result = store.nextForState(state.code(), 10);
-        assertThat(result).hasSize(1).allSatisfy(neg -> assertThat(neg).usingRecursiveComparison().isEqualTo(n));
-    }
-
-    @Test
-    void nextForState_lockEntity() {
-        var n = generateNegotiation("test-id-lock", ContractNegotiationStates.CONSUMER_OFFERED);
-        var doc = new ContractNegotiationDocument(n, partitionKey);
+        Duration leaseDuration = Duration.ofSeconds(5);
+        doc.acquireLease("another-connector", leaseDuration);
         container.createItem(doc);
 
-        // verify nextForState sets the lease
-        var result = store.nextForState(ContractNegotiationStates.CONSUMER_OFFERED.code(), 5);
-        assertThat(result).hasSize(1).extracting(ContractNegotiation::getId).containsExactly(n.getId());
-        var storedDoc = readItem(n.getId());
+        // before the lease expired
+        var negotiationsBeforeLeaseExpired = store.nextForState(state.code(), 10);
+        assertThat(negotiationsBeforeLeaseExpired).isEmpty();
```

Comment:
is this assertion really needed for this test ? on a very slow system it is possible that this check execute after 5 second and cause this assertion to fail.

I understand that normally this should not happen but just highlighting to see if we need this assertion or not, if not really needed then can be better to remove it.

And also similar check is already performed within test `nextForState_leasedByAnother`

Reply:
yes, indeed these two tests can be merged. It's good to see that at first the item is not available, it becomes available after the lease expired.

Reply:
I removed `nextForState_leasedByAnother` as it's covered in this test